build-support/bin/generate_docs.py

```diff
@@ -50,10 +56,14 @@ def create(cls) -> "ReferenceGenerator":
         parser = argparse.ArgumentParser(description="Generate the Pants reference markdown files.")
         default_output_dir = PosixPath(os.path.sep) / "tmp" / "pants_docs" / "help" / "option"
         parser.add_argument(
-            "--input",
-            default=None,
-            help="Path to a file containing the output of `./pants help-all`. "
-            "If unspecified, reads from stdin.",
+            "--help-input",
+            required=True,
+            help="Path to a file containing the output of `./pants help-all`.",
+        )
+        parser.add_argument(
```

Comment:
It occurs to me that this data should be in the help-all output JSON, kinda by definition of "all"...

Reply:
Do you think the `target-types` goal should still exist? I think yes, as `./pants target-types` is useful to list what's available.

It would be neat to be able to do `./pants help python_library`, though; but this would require that there are no collisions between subsystems and target types, which we don't enforce.

If we decide the `target-types` goal needs to still exist, I think it makes more sense to still have `target-types --all` than `help-all`, so that target info is accessed in a consistent way. Not a big deal either way.

Reply:
Yes, I do think target-types needs to exist. This is just for the json dump.

Reply:
Although it does occur to me that we might want to spell it `./pants help targets`...